Lesson6/src/com/home/military/MilitaryUnit.java

```diff
@@ -75,10 +75,10 @@ private boolean doesRecruitAlreadyExists(Person person, Person[] recruits) throw
         return false;
     }
 
-    public void infoRecrutsMilitaryUnit(MilitaryUnit militaryUnit) {
-        String[] namesRecruits = new String[militaryUnit.recruitsArray.length];
-        for (int i = 0; i < militaryUnit.recruitsArray.length; i++) {
-            namesRecruits[i] = militaryUnit.recruitsArray[i].getSurname() + " " + militaryUnit.recruitsArray[i].getName();
+    public void infoRecrutsMilitaryUnit() {
+        String[] namesRecruits = new String[capacity-getFreePlacesFromArray()];
+        for (int i = 0; i < capacity-getFreePlacesFromArray(); i++) {
```

Comment:
code formatting (spaces before and after the minus sign)

Reply:
also, you can just use takenPlacesCount instead of capacity-getFreePlacesFromArray()